docs/DMS/PRIMARY-DATA-STORAGE/README.md

```diff
@@ -369,3 +369,157 @@ Transaction:
 
 Use the SQL statements in the proof-of-concept code. Be sure to use the `VARCHAR` version
 rather than `VARBINARY`. Do not apply any foreign key cascades in the initial work.
+
+## Alternative Table Designs
+
+> [!NOTE]
+> The following designs and text were developed by GitHub Copilot using
+> GPT-5.1-Codex, based on the following prompt, with this file itself as
+> additional context: "Analyze this design document and offer alternative table
+> designs that would meet the objectives."
+
+Each option below preserves deterministic referential IDs, tunable partitioning, and the requirement that a
+single transaction envelope spans every write. They offer different trade-offs for existence validation,
+query fan-out, and operational isolation.
+
+### Partitioned Doc Store with Reference Edge Table
+
+This variant keeps the single `Documents` table but decomposes the `References` responsibilities into a
+lightweight `ReferenceEdges` table plus an `AliasLookup`. Reference existence checks rely on a single
+covering index over `AliasLookup(ReferentialPartitionKey, ReferentialId)`, so inserts do not need to join
+back to `Documents` immediately. That improves write throughput, allows cache-friendly lookups per
+partition, and still enforces referential integrity for delete/update flows.
+
+```mermaid
+erDiagram
+  Documents ||--o{ AliasLookup : "Document exposes referential IDs"
+  Documents ||--o{ ReferenceEdges : "Parent document edges"
+  AliasLookup ||--o{ ReferenceEdges : "Edges target referential id"
+  Documents {
+    bigint Id PK "Sequential, partition-aligned"
+    tinyint DocumentPartitionKey "Hash of DocumentUuid"
+    uuid DocumentUuid "API id, unique"
+    json EdfiDoc "Compressed payload"
+  }
+  AliasLookup {
+    uuid ReferentialId PK "Deterministic identity"
```

Comment:
The big downside to this design. uuid as a PK and an FK in the other table. We've gotten rid of that in the current design. Separately, ReferenceEdges needs the partition key for AliasLookup, but that's minor.